### ISSUE TITLE
Disabling one window input and hiding other window leads to nothing being drawn

### ISSUE THREAD

**Issue by Predelnik** [vurtun/nuklear#558]
So I've made very simple (I hope) example. It has 3 windows, presing button on the first window should hide second window and attach `NK_WINDOW_NO_INPUT` to the third.
https://gist.github.com/Predelnik/cf946fb7ad0b0efc2c6c98ee4e80fa2f
Result is nothing(including the first window) is drawn at all after pressing the button.

Looks like the cause is in

         if (!(win->flags & NK_WINDOW_HIDDEN) && !(win->flags & NK_WINDOW_NO_INPUT))

in `nk_begin_titled` which ends up not calling `nk_start` for window with disabled input which ends up not calling `nk_start_buffer` though buffer reallocation might be needed due to hiding of the window.
Looks like removing condition for `NK_WINDOW_NO_INPUT` resolves it so if it's just for optimization maybe that's the way to go but that's up to you of course. Thank you in advance!

**Comment by vurtun** [comment]
Hey @Predelnik ,
thanks for the issue. I tested it and if I click the button of the first window the third window will be hidden and the second window is not clickable anymore. If I click the button again the third window will reappear and the second window is clickable again. Did you make sure you are compiling with an up to date version of nuklear. I fixed a similar bug quite recently. 

My test code:
```c
        /* GUI */
        if (nk_begin(ctx, "Demo1", nk_rect(50, 50, 200, 200),
            NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_SCALABLE|
            NK_WINDOW_CLOSABLE|NK_WINDOW_MINIMIZABLE|NK_WINDOW_TITLE))
        {
            enum {EASY, HARD};
            static int op = EASY;
            static int property = 20;

            nk_layout_row_static(ctx, 30, 80, 1);
            if (nk_button_label(ctx, "button"))
                button_pressed = !button_pressed;
            nk_layout_row_dynamic(ctx, 30, 2);
            if (nk_option_label(ctx, "easy", op == EASY)) op = EASY;
            if (nk_option_label(ctx, "hard", op == HARD)) op = HARD;
            nk_layout_row_dynamic(ctx, 22, 1);
            nk_property_int(ctx, "Compression:", 0, &property, 100, 10, 1);
        }
        nk_end(ctx);

        int flags = NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_SCALABLE|
            NK_WINDOW_CLOSABLE|NK_WINDOW_MINIMIZABLE|NK_WINDOW_TITLE;
        if (button_pressed)
            flags |= NK_WINDOW_NO_INPUT;
        if (nk_begin(ctx, "Demo2", nk_rect(250, 50, 200, 200), flags))
        {
            enum {EASY, HARD};
            static int op = EASY;
            static int property = 20;

            nk_layout_row_static(ctx, 30, 80, 1);
            if (nk_button_label(ctx, "button"))
                fprintf(stdout, "button pressed\n");
            nk_layout_row_dynamic(ctx, 30, 2);
            if (nk_option_label(ctx, "easy", op == EASY)) op = EASY;
            if (nk_option_label(ctx, "hard", op == HARD)) op = HARD;
            nk_layout_row_dynamic(ctx, 22, 1);
            nk_property_int(ctx, "Compression:", 0, &property, 100, 10, 1);
        }
        nk_end(ctx);

        nk_window_show (ctx, "Demo3", button_pressed ? NK_HIDDEN : NK_SHOWN);
        if (nk_begin(ctx, "Demo3", nk_rect(450, 50, 200, 200),
            NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_SCALABLE|
            NK_WINDOW_CLOSABLE|NK_WINDOW_MINIMIZABLE|NK_WINDOW_TITLE))
        {
            enum {EASY, HARD};
            static int op = EASY;
            static int property = 20;

            nk_layout_row_static(ctx, 30, 80, 1);
            if (nk_button_label(ctx, "button"))
                fprintf(stdout, "button pressed\n");
            nk_layout_row_dynamic(ctx, 30, 2);
            if (nk_option_label(ctx, "easy", op == EASY)) op = EASY;
            if (nk_option_label(ctx, "hard", op == HARD)) op = HARD;
            nk_layout_row_dynamic(ctx, 22, 1);
            nk_property_int(ctx, "Compression:", 0, &property, 100, 10, 1);
        }
        nk_end(ctx);
```
I used the SDL2 demo, but that shouldn't make a difference.

**Comment by Predelnik** [comment]
Ok first of all I assure you that I'm using the latest master and behavior you are describing is correct and expected

I rechecked my sample and as it happens the bug surprisingly shows itself on it only in optimized build by MSVC. It's quite curious, however maybe it goes to show the undefined behavior is in the nature of this bug and not the bug in Microsoft optimizer but I'm not exactly sure.

I came out with the following recipee to make it reproducable on unoptimized MSVC and linux optimized builds:
Take whole code block related to "Demo2" window and place it below the code block for "Demo3" window, effectively swapping them. I hope my description is clear.
